File: Code/0001_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> numsearch;
        vector<int> res;
        res.resize(2);

        for(auto i = 0; i < nums.size(); i++)
        {
            if(numsearch.find(nums[i]) != numsearch.end())
            {
                res[0] = numsearch[nums[i]];
                res[1] = i;

                return res;
            }
            else
            {
                int r = target - nums[i];
                numsearch[r] = i;
            }
        }

        return res;
    }
};

File: Code/0001_Two_Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        List<Integer> list = new ArrayList<Integer>();
        
        for(int i = 0; i < nums.length; i++)
        {
            for(int j = i + 1; j < nums.length; j++)
            {
                if(nums[i] + nums[j] == target)
                {
                    list.add(i);
                    list.add(j);
                }
            }
        }
        
        int[] indices = new int[list.size()];
        for(int i = 0; i < list.size(); i++)
        {
            indices[i] = list.get(i);
        }
        
        return indices;
    }
}